src/dotnet-roslyn/VS/VSBranchInfo.cs

```diff
@@ -0,0 +1,154 @@
+        var buildConnection = dnceng;
+
+        if (builds is null or { Count: 0 })
+        {
+            throw new Exception($"Couldn't find build for build number: {buildNumber}");
+        }
+
+        foreach (var build in builds)
+        {
+            WriteNameAndValue("Build Number", build.BuildNumber);
+            WriteNameAndValue("Commit SHA", build.SourceVersion);
+            WriteNameAndValue("Link", $"{product.RepoBaseUrl}/tree/{build.SourceVersion}", "    ");
+            WriteNameAndValue("Source Branch", build.SourceBranch.Replace("refs/heads/", ""));
+            WriteNameAndValue("Build", ((ReferenceLink)build.Links.Links["web"]).Href);
+
+            if (showArtifacts)
+            {
+                await WriteArtifactInfo(product, buildConnection, build);
+            }
+        }
+    }
+
+    // Inspired by Mitch Denny: https://dev.azure.com/mseng/AzureDevOps/_git/ArtifactTool?path=/src/ArtifactTool/Commands/PipelineArtifacts/PipelineArtifactDownloadCommand.cs&version=GBusers/midenn/fcs-integration&line=68&lineEnd=69&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents
+    // Note: He wishes not to have his name attached to it.
+    private static async Task WriteArtifactInfo(IProduct product, AzDOConnection connection, Build build)
+    {
+        if (product.ArtifactsFolderName is null || product.ArtifactsSubFolderNames.Length == 0)
+            return;
+
+        var artifact = await connection.BuildClient.GetArtifactAsync(build.Project.Id, build.Id, product.ArtifactsFolderName);
+
+        var (id, name) = ParseContainerId(artifact.Resource.Data);
+
+        var projectId = (await connection.ProjectClient.GetProject(connection.BuildProjectName)).Id;
+
+        var items = await connection.ContainerClient.QueryContainerItemsAsync(id, projectId, name);
+
+        var links = from i in items
+                    where i.ItemType == VisualStudio.Services.FileContainer.ContainerItemType.Folder
+                    where product.ArtifactsSubFolderNames.Contains(i.Path)
+                    select (i.Path.Split('/').Last(), i.ContentLocation + "&%24format=zip&saveAbsolutePath=false"); // %24 == $
+
+        WriteNamesAndValues("Packages", links);
+
+        static (long id, string name) ParseContainerId(string resourceData)
+        {
+            // Example of resourceData: "#/7029766/artifacttool-alpine-x64-Debug"
+            var segments = resourceData.Split('/');
+
+            if (segments.Length == 3 && segments[0] == "#" && long.TryParse(segments[1], out var containerId))
+            {
+                return (containerId, segments[2]);
+            }
+
+            throw new Exception($"Resource data value '{resourceData}' was not in expected format.");
+        }
+    }
+
+    private static void WriteHeader(string branch)
+    {
+        Console.ForegroundColor = ConsoleColor.Cyan;
+        Console.WriteLine($"{branch}:");
+        Console.ResetColor();
```

Comment:
Could use the `logger` for writing your output, although you would lose direct control over the color without additional work.

Reply:
I thought about that, but then when writing out the info, I would never know if the user can actually see it, or if they passed `-v minimal` and are wondering why the app doesn't do anything :)

Reply:
you never can tell with users =) anyhoo, looks good to me